File: Managio-API/Models/UserProject.cs
using System.ComponentModel.DataAnnotations;

namespace Managio_API.Models
{
    public class UserProject
    {
        [Key]
        public int UserId { get; set; }

        [Key]
        public int ProjectId { get; set; }


        public virtual User User { get; set; }
        public virtual Project Project { get; set; }
    }
}

File: Mangio-SPA/src/app/_models/userProject.ts
import { User } from './user';
import { Project } from './project';

export interface UserProject {
    userId: number;
    projectId: number;
    user: User;
    project: Project
    projectNumber: string;
    projectName: string;
    isFinished: boolean;
    firstName: string;
    lastName: string;
}
